### PR TITLE
170349161 pushpin sync

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
WIP, all working aside from the VR interactions, which needs a bit of work tomorrow morning to get the pins working. The network is working well locally, though I notice if you launch a scene without going to LoadSim first, you get put in a random group - I think, for testing, it might be good to have a specific group name set behind the scenes, or a predictable group to join, so we don't end up wondering why our traffic isn't arriving :) Network PR is this one - added the small changes to pins today https://github.com/concord-consortium/CEASAR-server/pull/10

### CODE REVIEW

File: Assets/Scripts/DataController.cs [hunk 1]
```diff
@@ -277,7 +269,19 @@ public void UpdateOnSceneLoad()
...
-        if (showHorizonView) positionNCP();
+        
+        // if we're in Horizon view, set location and date / time
+        if (showHorizonView)
+        {
+            if (manager.UserHasSetLocation)
+            {
+                nextLocation = manager.LocalUserPin.Location;
+                // Force an update once ready
+                currentLocation = new LatLng();
```
Comment (knowuh) [inline Assets/Scripts/DataController.cs]: Seems like we need to  set time here eventually.

File: Assets/Scripts/InteractionController.cs [hunk 2]
```diff
@@ -158,13 +186,19 @@ public void ShowEarthMarkerInteraction(Vector3 pos, Quaternion rot, Color player
...
-            SimulationEvents.GetInstance().LocationChanged.Invoke(latLng, "Custom: ");
+            SimulationEvents.GetInstance().LocationChanged.Invoke(latLng, SimulationConstants.CUSTOM_LOCATION);
             CCLogger.Log(CCLogger.EVENT_ADD_INTERACTION, interactionInfo);
         }
     }
     public void SetEarthLocationPin(Vector3 pos, Quaternion rot, Color playerColor, bool isLocal)
     {
-        Vector2 latLng = Vector2.zero; // unset value
+        LatLng latLng = getEarthRelativeLatLng(pos);
+        AddOrUpdatePin(latLng, playerColor, manager.LocalUsername, manager.CurrentSimulationTime, true, true);
```
Comment (knowuh) [inline Assets/Scripts/InteractionController.cs]: Is `manager.LocalUsername` correct? If so, it seems like we could remove `playerColor` from the argument list, and also `isLocal` argument.

File: Assets/Scripts/InteractionDetect.cs [hunk 3]
```diff
@@ -101,5 +124,37 @@ void Update()
                 }
             }
         }
+        // else
```
Comment (knowuh) [inline Assets/Scripts/InteractionDetect.cs]: I assume that all this commented out code is just in flux, so I am not reviewing it.

File: Assets/Scripts/MainUIController.cs [hunk 4]
```diff
@@ -475,10 +481,11 @@ public void ToggleRunSimulation()
     public void CreateSnapshot()
     {
         // get values from datacontroller
-        DateTime snapshotDateTime = dataController.CurrentSimUniversalTime;
-        String location = dataController.StartCity;
+        DateTime snapshotDateTime = SimulationManager.GetInstance().CurrentSimulationTime;
```
Comment (knowuh) [inline Assets/Scripts/MainUIController.cs]: I think we could use the gettter `manager` you setup ...

File: Assets/Scripts/Core/Pushpin.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+using System;
```
[the remark this answers is not in the document]
Comment (dstrawberrygirl) [reply Assets/Scripts/Core/Pushpin.cs]: One day I'll remember to use it :) it's a good technique!

File: Assets/Scripts/DataController.cs (hunk 1 above)
Comment (dstrawberrygirl) [reply Assets/Scripts/DataController.cs]: This _should_ be picked up from the manager's CurrentSimulationTime, though there are issues there to be resolved...

File: Assets/Scripts/InteractionController.cs (hunk 2 above)
Comment (dstrawberrygirl) [reply Assets/Scripts/InteractionController.cs]: There's definitely some tangled logic here - I'll attempt to resolve along with the datetime issues

**Pull request merged**